### PR TITLE
Feat/day3

### PR DESCRIPTION
**Pull request by daveingDave**

### CODE REVIEW

File: src/day02/SubmarinePosition.php [hunk 1]
```diff
@@ -0,0 +1,34 @@
+<?php declare(strict_types=1);
```
Comment (Trawais) [inline src/day02/SubmarinePosition.php:1]: The whole folder `day02` should not be in this pull-request.

File: src/day03/BitRates.php [hunk 2]
```diff
@@ -0,0 +1,78 @@
+<?php declare(strict_types=1);
+
+namespace AdventOfCode\day03;
+
+final class BitRates
+{
+    public function getBitRate(array $input): int
+    {
+        $gama = "";
```
Comment (Trawais) [inline src/day03/BitRates.php:9]: `gama` and `epsilon` is forgotten here from the first part of the task. Should be removed.

File: src/day03/day03Test.php [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+use PHPUnit\Framework\TestCase;
+
+final class day03Test extends TestCase
+{
+    public function testGivenExamplePasses(): void
+    {
+        $measurement = new BitRates();
+        $input = ["00100", "11110", "10110", "10111", "10101", "01111", "00111", "11100", "10000", "11001", "00010", "01010"
+        ];
```
Comment (Trawais) [inline src/day03/day03Test.php:13]: I would not keep the ending box bracket on the new line but on the same.

File: src/day03/BitRates.php [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+            $zero = 0;
+            $one = 0;
+            $oneArray = [];
+            $zeroArray = [];
+            $numberOfCharInString++;
+        }
+
+        $numberOfCharInString = 0;
+
+        while(count($co2Rating)>1) {
```
Comment (Trawais) [inline src/day03/BitRates.php:49]: Let's try to separate the same logic into only one function. One additional parameter must be added to decide whether you want to count ones or zeros.

Comment (Trawais) [reply src/day03/BitRates.php:49]: Then I believe this code could be much shorther.

**Review by Trawais — CHANGES_REQUESTED** [review]

**Review by Trawais — APPROVED** [review]

File: src/day03/BitRates.php [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+            }
+            $value = $this->returnRightValues($one, $zero, $oneArray, $zeroArray, $zeroOrOne);
+            $numberOfCharInString++;
+        }
+        return bindec($value[0]);
+    }
+
+    public function getBitRate(array $input): int
+    {
+        return $this->getOxygenGeneratorOrCo2Rating($input ,1) * $this->getOxygenGeneratorOrCo2Rating($input, 0);
```
Comment (Trawais) [inline src/day03/BitRates.php]: fix your code style (comma before `1`)

**Review by Trawais — APPROVED** [review]

**Review by Bollmanmkvb — APPROVED** [review]

**Pull request merged**